R/timeseries_plot.R

```diff
@@ -0,0 +1,103 @@
+#' Title
+#'
+#' @param data 
+#' @param date_time 
+#' @param elements 
+#' @param station 
+#' @param facets 
+#' @param add_points 
+#' @param add_line_of_best_fit 
+#' @param se 
+#' @param add_path 
+#' @param add_step 
+#' @param na.rm 
+#' @param show.legend 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+timeseries_plot <- function(data, date_time, elements, station = NULL, facets = c("stations", "elements", "both", "none"),
+                            add_points = FALSE, add_line_of_best_fit = FALSE,
+                            se = TRUE, add_path = FALSE, add_step = FALSE,
+                            na.rm = FALSE, show.legend = NA){
+  
+  checkmate::assert_data_frame(data)
+  # date_time can be a date, factor, or character.
+  checkmate::assert_character(elements)
+  checkmate::assert_character(station, null.ok = TRUE)
```

Comment:
Can also add `assert_column_names` check which is a function I've added in the package.